frontend/src/router/routeHandlers.ts

```diff
@@ -91,10 +91,30 @@ const beforeEnterGrantApplicationAdminRoute = async (
     return true;
 }
 
+const beforeEnterGrantDelegationAdminRoute = async (
+    to: RouteLocationNormalized
+) => {
+
+    if (selectedApplicationId.value === FAM_APPLICATION_ID) {
+        emitRouteToastError(ACCESS_RESTRICTED_ERROR);
+        return { path: routeItems.dashboard.path };
+    }
```

Comment:
Hi Nick, I think I missed one situation, we have to verify the user is the admin of the selected application. For example, I'm the delegated admin of FOM_DEV, app admin of FOM_TEST, when I selected FOM_DEV, even though I still have the app admin role, but I'm not the app admin of FOM_DEV, then I shouldn't be able to access to this page.